File: backend/app/schemas/quiz.py
from typing import List, Optional
from pydantic import BaseModel, Field


class QuizOption(BaseModel):
    """퀴즈 선택지"""
    id: str
    text: str


class QuizQuestion(BaseModel):
    """퀴즈 질문"""
    id: int
    question: str
    options: List[QuizOption]


class QuizQuestionsResponse(BaseModel):
    """퀴즈 질문 목록 응답"""
    questions: List[QuizQuestion]
    total_questions: int


class QuizAnswer(BaseModel):
    """개별 퀴즈 답변"""
    question_id: int = Field(..., ge=1, le=5)
    option_id: str = Field(..., pattern="^[a-d]$")


class QuizSubmitRequest(BaseModel):
    """퀴즈 제출 요청"""
    answers: List[QuizAnswer] = Field(..., min_length=5, max_length=5)


class QuizSubmitResponse(BaseModel):
    """퀴즈 제출 응답"""
    success: bool
    persona_id: int
    persona_name: str
    persona_level: str
    message: str


class PersonaLevelInfo(BaseModel):
    """페르소나 레벨 정보"""
    current_level: str
    level_name: str
    description: str
    diary_count: int
    next_level: Optional[str]
    next_level_name: Optional[str]
    diaries_needed: Optional[int]
    progress_percent: int

File: frontend/src/types/quiz.ts
// Quiz related types

export interface QuizOption {
  id: string
  text: string
}

export interface QuizQuestion {
  id: number
  question: string
  options: QuizOption[]
}

export interface QuizQuestionsResponse {
  questions: QuizQuestion[]
  total_questions: number
}

export interface QuizAnswer {
  question_id: number
  option_id: string
}

export interface QuizSubmitRequest {
  answers: QuizAnswer[]
}

export interface QuizSubmitResponse {
  success: boolean
  persona_id: number
  persona_name: string
  persona_level: string
  message: string
}

export interface PersonaLevelInfo {
  current_level: string | null
  level_name: string
  description: string
  diary_count: number
  next_level: string | null
  next_level_name: string | null
  diaries_needed: number | null
  progress_percent: number
}
